lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -236,6 +260,30 @@ pub fn rectangle_hashes(rectangle: &GeoBoundingBox, max_regions: usize) -> Vec<G
         .expect("geo-hash coverage is empty")
 }
 
+/// Return as-high-as-possible with maximum of `max_regions`
+/// number of geo-hash guaranteed to contain the whole polygon.
+pub fn polygon_hashes(polygon: &GeoPolygon, max_regions: usize) -> Vec<GeoHash> {
+    assert_ne!(max_regions, 0, "max_regions cannot be equal to zero");
+    let geo_bounding_box = minimum_bounding_rectangle_for_polygon(polygon);
+    let full_geohash_bounding_box: GeohashBoundingBox = geo_bounding_box.into();
+
+    (0..=GEOHASH_MAX_LENGTH)
+        .map(|precision| {
+            full_geohash_bounding_box
+                .geohash_regions(precision, max_regions)
+                .map(|hashes| {
+                    hashes
+                        .into_iter()
+                        .filter(|hash| check_polygon_intersection(hash, polygon))
+                        .collect_vec()
+                })
+        })
+        .take_while(|hashes| hashes.is_some())
+        .last()
+        .expect("no hash coverage for any precision")
+        .expect("geo-hash coverage is empty")
```

Comment:
We prefer to propagate errors rather than panicking

Reply:
Thanks for pointing out :)  I've noticed that the functions`circle_hashes`, `polygon_hashes`, and `rectangle_hashes` currently trigger panic instead of throwing exceptions upward. Consequently, I plan to create an additional patch to modify this behavior in all three functions while ensuring that this commit remains simple. Do you think this approach is suitable?

Reply:
It's true, I wasn't familiar with this part of the codebase. I like what you're suggesting